### DISCUSSION TITLE
[PYTHON] saving script results as artifact

### DISCUSSION THREAD

**Discussion by shaibrown5** [community/community#26577] [Actions]
<p>I have a python script that runs an MLOps pipeline, and in the end, I get the accuracy.</p>
<p>I want to put the accuracy in a file and upload it as an artifact during the workflow to the github actions.</p>
<p>I know how to upload an artifact, the problem is writing a file in python during the workflow run that holds the information I want. It is important to point out that the script does not return the accuracy.</p>
<p>what I have tried until now is using ‘with open(file.txt, ‘w’)…’ and writing to it. however, when I try to upload file.txt as an artifact, I attempt to upload from ${{github.workspace}}/path/to/file/file.txt, the workflow can not find it</p>
<p>After a couple of attempts, and using “run: ls ${{github.workspace}}/path/to/file/” I have noticed that it does not even create the file, nor does it write to it.</p>
<p>I have tried creating the file as an individual step in the workflow, and then writing to it, but writing to it during the python script does not work.</p>
<p>Is there any way that I can retrieve the accuracy information from the python script without returning it in the method?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I wonder if the path might be wrong? If you do <code>open('file.txt', 'w')</code> the file will appear in the current working directory, not the directory where the script file is.</p>

**Comment by shaibrown5** [comment]
<p>so are you suggesting it may be under ${{github.workspace}}/ ?</p>
<p>I will run a test now, and print out the directory to see if it is there</p>

**Comment by airtower-luna** [comment]
<p>If you didn’t change the default working directory that’s where it should be, yes. :slightly_smiling_face:</p>
